src/a5dd0_len_114e0.c

```diff
@@ -2988,8 +2988,638 @@ void state_render_frontUI(void) {
+                case 0:
+                    gSPSetLights0((*gfxPos)++, lightningGroup[0]);
+                    break;
+                case 1:
+                    gSPSetLights1((*gfxPos)++, lightningGroup[0]);
+                    break;
+                case 2:
+                    gSPSetLights2((*gfxPos)++, lightningGroup[0]);
+                    break;
+                case 3:
+                    gSPSetLights3((*gfxPos)++, lightningGroup[0]);
+                    break;
+                case 4:
+                    gSPSetLights4((*gfxPos)++, lightningGroup[0]);
+                    break;
+                case 5:
+                    gSPSetLights5((*gfxPos)++, lightningGroup[0]);
+                    break;
+                case 6:
+                    gSPSetLights6((*gfxPos)++, lightningGroup[0]);
+                    break;
+                case 7:
+                    gSPSetLights7((*gfxPos)++, lightningGroup[0]);
+                    break;
+            }
+        }
+    }
+
+    if (textureHeader != NULL) {
+        switch (extraTileType) {
+            case EXTRA_TILE_3:
+            case EXTRA_TILE_4:
+                prop = get_model_property(modelNode, MODEL_PROP_KEY_SPECIAL);
+                if (prop != NULL) {
+                    s32 v1 = prop->data.s;
+                    u16 a2 = prop->dataType;
+                    s32 a1 = prop->dataType;
+                    func_801180E8(textureHeader, gfxPos, textureHandle->raster, textureHandle->palette, textureHandle->auxRaster, textureHandle->auxPalette,
+                                (v1 >> 12) & 0xF, (v1 >> 16) & 0xF,
+                                a2 & 0xFFF, (a1 >> 12) & 0xFFF);
+
+                } else {
+                    gSPDisplayList((*gfxPos)++, textureHandle->gfx);
+                }
+                break;
+            default:
+                gSPDisplayList((*gfxPos)++, textureHandle->gfx);
+                break;
+        }
+    } else {
+        gSPTexture((*gfxPos)++, 0, 0, 0, G_TX_RENDERTILE, G_OFF);
+        gDPSetCombineMode((*gfxPos)++, G_CC_SHADE, G_CC_SHADE);
+        gDPSetColorDither((*gfxPos)++, G_CD_MAGICSQ);
+        gDPSetAlphaDither((*gfxPos)++, G_AD_PATTERN);
+    }
+
+    if (combineSubType != 0 || renderMode == RENDER_MODE_ALPHATEST || renderMode == RENDER_MODE_ALPHATEST_ONESIDED) {
+        u32 v1 = 0;
+
+        if (textureHeader != NULL) {
+            u32 colorCombineType = textureHeader->colorCombineType;
+            if (colorCombineType >= 3) {
+                v1 = colorCombineType + 10;
+            } else {
+                v1 = extraTileType * 3 + 1 + textureHeader->colorCombineSubType;
+            }
+        }
+
+        if (renderMode != RENDER_MODE_ALPHATEST && renderMode != RENDER_MODE_ALPHATEST_ONESIDED) {
+            *(*gfxPos) = D_8014B0B8[v1][combineSubType];
+        } else {
+            *(*gfxPos) = D_8014B400[v1][combineSubType];
+        }
+        (*gfxPos)++;
+    }
+
+    switch (s3) {
+        case 1:
```

Comment:
This is some 'texturing mode' enum, with at least these values:
1 = NO_TEXTURE
2 = NORMAL
3 = FOG
6 = FOG_MULTIPLY_COLOR